lib/openscap_parser/value.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+module OpenscapParser
+  class Value < XmlNode
+    include OpenscapParser::Util
+
+    def id
+      @id ||= parsed_xml['id']
+    end
+
+    def description
+      @description ||= newline_to_whitespace(parsed_xml.at_css('description')&.text)
+    end
+
+    def title
+      @title ||= parsed_xml.at_css('title')&.text
+    end
+
+    def type
+      @type ||= parsed_xml['type'] || 'string'
+    end
+
+    def lower_bound(selector = nil)
+      @lower_bounds ||= parsed_xml.xpath('lower-bound').each_with_object({}) do |element, lb|
+        lb[element.at_xpath('@selector')&.text.presence] = element&.text
+      end
+
+      return @lower_bounds[selector] if selector
+      @lower_bounds[nil]
+    end
+
+    def upper_bound(selector = nil)
+      @upper_bounds ||= parsed_xml.xpath('upper-bound').each_with_object({}) do |element, ub|
+        ub[element.at_xpath('@selector')&.text.presence] = element&.text
+      end
+
+      return @upper_bounds[selector] if selector
+      @upper_bounds[nil]
+    end
+
+    def value(selector = nil)
+      @values ||= parsed_xml.xpath("value").each_with_object({}) do |element, rv|
+        rv[element.at_xpath('@selector')&.text.presence] = element&.text
+      end
+
+      return @values[selector] if selector
+      @values[nil] || @values.values.first
+    end
```

Comment:
These are 3 copy-pasted methods...

Reply:
Hmm.. I don't see the problem 🤔 they're similar because they do similar things

Reply:
DRY

Reply:
Ohh right I can combine them into one